src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -0,0 +1,469 @@
+            ) if base_image_trigger_enabled else None
+        )
+    )
+
+    try:
+        return client.create(resource_group_name=resource_group_name,
+                             registry_name=registry_name,
+                             task_name=task_name,
+                             task_create_parameters=task_create_parameters)
+    except ValidationError as e:
+        raise CLIError(e)
+
+
+def acr_task_show(cmd,
+                  client,
+                  task_name,
+                  registry_name,
+                  with_secure_properties=False,
+                  resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd.cli_ctx, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    if with_secure_properties:
+        return client.get_details(resource_group_name, registry_name, task_name)
+    return client.get(resource_group_name, registry_name, task_name)
+
+
+def acr_task_list(cmd,
+                  client,
+                  registry_name,
+                  resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd.cli_ctx, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+    return client.list(resource_group_name, registry_name)
+
+
+def acr_task_delete(cmd,
+                    client,
+                    task_name,
+                    registry_name,
+                    resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd.cli_ctx, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+    return client.delete(resource_group_name, registry_name, task_name)
+
+
+def acr_task_update(cmd,  # pylint: disable=too-many-locals
+                    client,
+                    task_name,
+                    registry_name,
+                    resource_group_name=None,
+                    # task parameters
+                    status=None,
+                    os_type=None,
+                    cpu=None,
+                    timeout=None,
+                    context_path=None,
+                    commit_trigger_enabled=None,
+                    git_access_token=None,
+                    branch=None,
+                    image_names=None,
+                    no_push=None,
+                    no_cache=None,
+                    file=None,
+                    values=None,
+                    arg=None,
+                    secret_arg=None,
+                    set_value=None,
+                    set_secret=None,
+                    base_image_trigger_enabled=None,
+                    base_image_trigger_type=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd.cli_ctx, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    task = client.get(resource_group_name, registry_name, task_name)
+    step = task.step
+    if isinstance(step, DockerBuildStep):
+        if file and file.endswith(ALLOWED_TASK_FILE_TYPES):
```

Comment:
Why do we throw if file ends with allowed file types?

Reply:
Earlier, we didn't want to allow updating step to a different type than what was created. That changed today morning, have pushed changes.